lib/omnibus-ctl.rb

```diff
@@ -506,6 +512,67 @@ def reconfigure(exit_on_success=true)
       end
     end
 
+    def check_license_acceptance(override_accept = false)
+      license_guard_file_path = File.join(data_path, ".license.accepted")
+
+      # If the project does not have a license we do not have
+      # any license to accept.
+      return unless File.exist?(project_license_path)
+
+      if !File.exist?(license_guard_file_path)
+        if override_accept || ask_license_acceptance
+          FileUtils.touch(license_guard_file_path)
+        else
+          puts "Please accept software's license in order to continue."
+          exit(1)
+        end
+      end
+    end
+
+    def ask_license_acceptance
+      puts "To use this software, you must agree to the terms of the software license agreement."
+
+      if !STDIN.tty?
+        puts "You need to either view and agree to the license agreement or pass --accept-license."
+        exit(1)
+      end
+
+      loop do
+        puts "Press 'Return' to view the software license agreement, 'q' to quit."
```

Comment:
The reason we inform first and then show the license agreement is because we were concerned that users will not know what's happening when they are presented with the license agreement. 

See [this](https://github.com/chef/omnibus-ctl/pull/42#discussion_r58690635) comment.